cmake/modules/FindMaterialX.cmake

```diff
@@ -0,0 +1,139 @@
+#
+# TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+# All rights reserved.  See LICENSE.txt for license.
+#
+# Helper CMake module to Find MaterialX include dirs, libraries and document libraries
+# Based on https://github.com/PixarAnimationStudios/USD/blob/master/cmake/modules/FindMaterialX.cmake
+#
+#
+# Example Usage
+#  - Build and Install MaterialX 
+#  - register the path to this cmake module using CMAKE_MODULE_PATH
+#  - use find_package(MaterialX REQUIRED)
+#
+# Variables defined:
+# MATERIALX_FOUND            True if headers and requested libraries were found
+# MATERIALX_BASE_DIR         MaterialX root installation directory
+# MATERIALX_INCLUDE_DIRS     MaterialX include directory
+# MATERIALX_LIB_DIRS         MaterialX lib directory
+# MATERIALX_CORE_LIBS        MaterialX Core libraries and ShaderX i.e. Core, Format & GenShader
+# MATERIALX_GENERATOR_LIBS   MaterialX Generator libraries i.e GenGlsl, GenOsl
+# MATERIALX_STDLIB_DIR       Path to the MaterialX standard library directory
+# MATERIALX_PBRLIB_DIR       Path to the MaterialX pbr library directory
+#
```

Comment:
I assume you'd also want to be able to find the Python modules "MATERIALX_PYTHON_DIR" ?

Reply:
One other comment, I guess is to allow resources/ to be found as well.